pkg/autoscaler/metrics/http_scrape_client_test.go

```diff
@@ -82,64 +107,72 @@ func TestNewHTTPScrapeClient_ErrorCases(t *testing.T) {
 }
 
 func TestHTTPScrapeClientScrapeHappyCase(t *testing.T) {
-	hClient := newTestHTTPClient(getHTTPResponse(http.StatusOK, testFullContext), nil)
-	sClient, err := newHTTPScrapeClient(hClient)
-	if err != nil {
-		t.Fatalf("newHTTPScrapeClient = %v, want no error", err)
-	}
+	for _, hClient := range []*http.Client{
+		newTestHTTPClient(getHTTPResponse(http.StatusOK, testFullContext), nil),
+		newTestHTTPClient(getProtoHTTPResponse(http.StatusOK, statFull, protoAcceptContent), nil),
+	} {
+		sClient, err := newHTTPScrapeClient(hClient)
+		if err != nil {
+			t.Fatalf("newHTTPScrapeClient = %v, want no error", err)
+		}
+		stat, err := sClient.Scrape(testURL)
+		if err != nil {
+			t.Errorf("scrapeViaURL = %v, want no error", err)
+		}
+		if stat.AverageConcurrentRequests != 3.0 {
```

Comment:
the numbers in the rest of this test (and TestHTTPScrapeClientScrapeHappyCaseWithOptionals) can be replaced with the defined constants